hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancerSelectionCriteria.java

```diff
@@ -68,6 +69,7 @@ public ContainerBalancerSelectionCriteria(
     this.replicationManager = replicationManager;
     this.containerManager = containerManager;
     selectedContainers = new HashSet<>();
+    excludeContainersDueToFailure = new HashSet<>();
```

Comment:
excludeContainersDueToFailure holds failure in the particular iteration, and choosing the source again avoiding the container.
But in next iteration, when same source is re-considered, IMO, this list also needs cleanup. This is required to be done in,
org.apache.hadoop.hdds.scm.container.balancer.ContainerBalancerTask#initializeIteration

Reply:
In ContainerBalancerTask#initializeIteration, in the end we are creating a new object for containerBalancerSelectionCriteria
```
selectionCriteria = new ContainerBalancerSelectionCriteria(config,
        nodeManager, replicationManager, containerManager, findSourceStrategy);
```
In the constructor, we are initializing `excludeContainersDueToFailure` to empty set. That should take care of this